pkgs/applications/video/dvb-apps/default.nix

```diff
@@ -1,14 +1,21 @@
-{ stdenv, fetchurl, perl }:
+{ stdenv, fetchurl, perl, pkgsi686Linux }:
 
-stdenv.mkDerivation {
-  name = "dvb-apps-7f68f9c8d311";
+pkgsi686Linux.stdenv.mkDerivation {
```

Comment:
`pkgsi686Linux` is an odd choice here. It restricts it to i686 only!? Just `stdenv.mkDerivation` should be fine here.

Reply:
I tried, but it was compiling with errors. This is the only way that worked and I didn't have much time to inspect the issues better.

Reply:
It is `0003-handle-static-shared-only-build.patch` which causes that problem. If you remove that patch it compiles with `stdenv.mkDerivation`. We do not really need that patch anyway. It would be nice if the build system honors static/shared but it does not hurt here to have both static and shared libs in the same output..